### ISSUE TITLE
tournament crash can--in theory--take down mb2

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#288]
If a tournament panics, it _can_ take down all of mb2. This appears to be due to mailboxes getting full, although I haven't looked at it closely. It happened most recently when I was stress-testing and ran into a bug when I quit the sole player waiting at a paused table (#287).

Tournaments are run in separate threads, so with a little work we can make it so that a crashed tournament never takes down the entire system. As it is, I don't think they usually do.

In the near future, my plan is to simply fix any panic that I see. However, after BARGE, I should boost the priority of this and maybe even label it as a bug.

**Comment by ctm** [comment]
I added a tiny change that changes a `panic!` to a `warn!` in a situation where one panic would cause an endless set of panics thereafter. It's a step in the right direction but is not sufficient for me to close this ticket.

FWIW, I added the change because I had accidentally written an invalid value to the db as I was working on ring games (#88) and when I started mb2 up I got the endless panics that I've seen before, so my fix definitely is an improvement.
